Source/qol/stash.cpp

```diff
@@ -0,0 +1,610 @@
+
+	if (StashButtonPressed != -1) {
+		Point stashButton = GetPanelPosition(UiPanels::Stash, StashButtonRect[StashButtonPressed].position);
+		DrawArt(out, stashButton, &StashNavButtonArt, StashButtonPressed);
+	}
+
+	constexpr Displacement offset { 0, INV_SLOT_SIZE_PX - 1 };
+
+	for (auto slot : PointsInRectangleRange({ { 0, 0 }, { 10, 10 } })) {
+		if (Stash.stashGrids[Stash.page][slot.x][slot.y] != 0) {
+			InvDrawSlotBack(out, GetStashSlotCoord(slot) + offset, InventorySlotSizeInPixels);
+		}
+	}
+
+	for (auto slot : PointsInRectangleRange({ { 0, 0 }, { 10, 10 } })) {
+		if (Stash.stashGrids[Stash.page][slot.x][slot.y] == 0) {
+			continue; // No item in the given slot
+		}
+
+		uint16_t itemId = Stash.stashGrids[Stash.page][slot.x][slot.y] - 1;
+		if (Stash.stashList[itemId].position != slot) {
+			continue; // Not the first slot of the item
+		}
+
+		Item &item = Stash.stashList[itemId];
+		int frame = item._iCurs + CURSOR_FIRSTITEM;
+
+		const Point position = GetStashSlotCoord(item.position) + offset;
+		const auto &cel = GetInvItemSprite(frame);
+		const int celFrame = GetInvItemFrame(frame);
+
+		if (pcursstashitem == itemId) {
+			uint8_t color = GetOutlineColor(Stash.stashList[itemId], true);
+			CelBlitOutlineTo(out, color, position, cel, celFrame, false);
+		}
+
+		CelDrawItem(Stash.stashList[itemId], out, position, cel, celFrame);
+	}
+
+	Point position = GetPanelPosition(UiPanels::Stash);
+	UiFlags style = UiFlags::VerticalCenter | UiFlags::ColorWhite;
+
+	DrawString(out, fmt::format("{:d}", Stash.page + 1), { position + Displacement { 132, 0 }, { 57, 11 } }, UiFlags::AlignCenter | style);
+	DrawString(out, fmt::format("{:d}", Stash.gold), { position + Displacement { 122, 19 }, { 107, 13 } }, UiFlags::AlignRight | style);
+}
+
+void CheckStashItem(Point mousePosition, bool isShiftHeld, bool isCtrlHeld)
+{
+	if (pcurs >= CURSOR_FIRSTITEM) {
+		CheckStashPaste(mousePosition);
+	} else {
+		CheckStashCut(mousePosition, isShiftHeld, isCtrlHeld);
+	}
+}
+
+uint16_t CheckStashHLight(Point mousePosition)
+{
+	Point slot = InvalidStashPoint;
+	for (auto point : PointsInRectangleRange({ { 0, 0 }, { 10, 10 } })) {
+		Rectangle cell {
+			GetStashSlotCoord(point),
+			{ InventorySlotSizeInPixels.width + 1, InventorySlotSizeInPixels.height + 1 }
+		};
+
+		if (cell.Contains({ mousePosition })) {
+			slot = point;
+			break;
+		}
+	}
+
+	if (slot == InvalidStashPoint)
+		return -1;
+
+	InfoColor = UiFlags::ColorWhite;
+
+	ClearPanel();
+
+	uint16_t itemId = abs(Stash.stashGrids[Stash.page][slot.x][slot.y]);
+	if (itemId == 0)
+		return -1;
+
+	uint16_t ii = itemId - 1;
+	Item &item = Stash.stashList[ii];
+
+	if (item.isEmpty())
+		return -1;
+
+	InfoColor = item.getTextColor();
+	if (item._iIdentified) {
+		strcpy(infostr, item._iIName);
+		PrintItemDetails(item);
+	} else {
+		strcpy(infostr, item._iName);
+		PrintItemDur(item);
+	}
+
+	return ii;
+}
+
+bool UseStashItem(uint16_t c)

```

Comment:
This is only used by `CtrlUseStashItem()`, which is itself unused.

Reply:
Yeah, it's going to be needed once we implement using items directly from the stash.